Adding @react-native-community/google-signin causes Android app to fail to load.

<!--
Please make sure you have searched previous issues before opening a new issue.

### If you're having a configuration error

Configuring google sign in can sometimes be tricky. If you're hitting a wall, you can get in touch with a maintainer ([@vonovak](https://github.com/vonovak)) via his [personal site](https://react-native-training.eu/). Please note that this is a paid service.

-->

<!-- Describe your issue in detail. -->
Adding `@react-native-community/google-signin` causes Android application to fail to open. And I am unable to find the steps I had to take to fix it documented.

(I am new to RN and Firebase and this package. I apologise in advance if I do not provide the right info in this issue - please let me if there is anything else I need to add.)

## Steps to Reproduce

(Omitted from the steps below were many invocations of `gradlew clean`, and many restarts/wipe data of the Android emulator.)

1. Create project codebase

`npx @react-native-community/cli@4.10.0 init --template=@react-native-firebase/template@6.8.1 MyApp
`

2. Run `gradlew signingReport` to generate SHA1.

3. Log into Firebase console and add new Android application to Firebase project.

4. Download `google-services.json` to `MyApp/android/app`

5. Enable sign-in methods: Email/Password, Google, Anonymous.

6. `npx react-native run-android` (SUCCESS!! 😄  Application loads in Android emulator)

7. `npm i @react-native-firebase/auth@6.7.1`

8. `npx react-native run-android` (SUCCESS AGAIN!! 😄 Application loads in Android emulator)

9. `npm i @react-native-community/google-signin`

10. `npx react-native run-android` ( 💥 FAILS TO OPEN!! Application fails to start in emulator possibly due to: `java.lang.ClassNotFoundException: com.facebook.flipper.ReactNativeFlipper`).

11. Open `MainApplication.java` and comment out the following line:

````
initializeFlipper(this); // Remove this line if you don't want Flipper enabled
````

10. `npx react-native run-android` ( 😞 STILL FAILS TO OPEN!! Application fails to start in emulator possibly due to: `Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.swiperefreshlayout.widget.SwipeRefreshLayout"`)

12. Add the following to `build.gradle`:
````
implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0-alpha02'
````

13. `npx react-native run-android` (😄 SUCCESS !! Opens)

## Expected Behavior

<!-- Write what you thought would happen. -->
By adding `@react-native-community/google-signin` I expected the application to still load.
I expected the documentation to include the steps I had to take to fix this.

## Actual Behavior

<!-- Write what happened. Include a gif if needed. If this is a regression, let us know. -->
By simply adding `@react-native-community/google-signin` as an NPM dependency the application fails to load in Android. `logcat` is the only way to find out why it failed to load.

## Environment

Package.json:
````
  "dependencies": {
    "@react-native-community/google-signin": "^3.0.4",
    "@react-native-firebase/app": "6.7.1",
    "@react-native-firebase/auth": "^6.7.1",
    "react": "16.9.0",
    "react-native": "0.61.5"
  },
````
The emulator was running Android 9.0 (Google Play)

Please let me know if there is any further information I can provide.

Thank you for building and maintaining this package!

--- a/docs/android-guide.md
+++ b/docs/android-guide.md
@@ -6,7 +6,7 @@ Please see the **FAQ** at bottom before opening new issues
 
 - Follow [this](./get-config-file.md) guide to get the configuration file.
 
-- Place the generated configuration file (`google-services.json`) into `<YOUR_PROJECT_ROOT>/android/app`
+- Place the generated configuration file (`google-services.json`) into project according to [this guide](https://developers.google.com/android/guides/google-services-plugin#adding_the_json_file).
 
 ### 2. Installation
 
@@ -57,7 +57,7 @@ dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
     implementation "com.android.support:appcompat-v7:23.0.1"
     implementation "com.facebook.react:react-native:+"
-    implementation(project(":react-native-community_google-signin")) // <--- add this dependency
+    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.0.0' // <-- add this; newer versions should work too
 }
 
 apply plugin: 'com.google.gms.google-services' // <--- this should be the last line
@@ -94,6 +94,17 @@ public class MainApplication extends Application implements ReactApplication {
 }
 ```
 
+in `android/app/build.gradle` you should have
+
+```gradle
+...
+dependencies {
+    ...
+    implementation(project(":react-native-google-signin"))
+}
+...
+```
+
 #### Choose Dependency versions (optional)
 
 The library depends on `com.google.android.gms:play-services-auth`, as seen in [build.gradle](https://github.com/react-native-community/google-signin/blob/master/android/build.gradle). If needed, you may control their versions by the `ext` closure, as seen in [build.gradle](https://github.com/react-native-community/google-signin/blob/master/example/android/build.gradle) of the example app.
@@ -102,11 +113,11 @@ The library depends on `com.google.android.gms:play-services-auth`, as seen in [
 
 Make sure you have a simulator with Google Play Services installed.
 
-Also to help with performance, install `HAXM` from the Android SDK Manager.
+To ensure best performance, you should use x86 emulator and have [HW acceleration](https://developer.android.com/studio/run/emulator-acceleration#accel-vm) working.
 
 ### Running on device
 
-Nothing special here, as long as you run your app on a Google Android device (again with Google Play Services installed!)
+Nothing special here, as long as you run your app on an Android device with Google Play Services installed.
 
 ## FAQ / Troubleshooting
 

